Add back claim clawback test

prajjwol opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: [#ENG-421](https://linear.app/tharsis/issue/ENG-421/keepergo-unit-tests-getter-and-setters)

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

prajjwol commented on x/claim/keeper/keeper_test.go:
```diff
@@ -75,6 +76,11 @@ func (suite *KeeperTestSuite) SetupTest() {
 }
 
+func init() {
+	config := sdk.GetConfig()
```
had to add back so `sdk.AccAddressFromBech32` in `claims_test.go` would work

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +454,65 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+	for _, tc := range tests {
+		addr, err := sdk.AccAddressFromBech32(tc.address)
+		suite.Require().NoError(err, "err: %s test: %s", err, tc.name)
+		acc := authtypes.NewBaseAccountWithAddress(addr)
```
nit, use the  EthAccount (or `AccountKeeper.NewAccount`)

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +454,65 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+		},
+		{
+			name:           "non airdrop address active",
+			address:        sdk.AccAddress(secp256k1.GenPrivKey().PubKey().Address()).String(),
```
you can use the etheremint's tests package to generate a valid ethereum address (we won't support the `secp256k1` keys at mainnet)

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +454,65 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+		suite.Require().NoError(err, "err: %s test: %s", err, tc.name)
+		coins := suite.app.BankKeeper.GetAllBalances(suite.ctx, addr)
+		if tc.expectClawback {
+			suite.Require().True(coins.AmountOf("aevmos").Equal(sdk.NewInt(0)),
```
```suggestion
			suite.Require().True(coins.AmountOfNoDenomValidation("aevmos").Equal(sdk.ZeroInt()),
```

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +454,65 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+			suite.Require().True(coins.AmountOf("aevmos").Equal(sdk.NewInt(0)),
+				"balance incorrect. test: %s", tc.name)
+		} else {
+			suite.Require().True(coins.AmountOf("aevmos").Equal(sdk.NewInt(100)),
```
```suggestion
			suite.Require().True(coins. AmountOfNoDenomValidation("aevmos").Equal(sdk.NewInt(100)),
```

fedekunze commented on x/claim/keeper/keeper_test.go:
```diff
@@ -75,6 +76,11 @@ func (suite *KeeperTestSuite) SetupTest() {
 	suite.app.StakingKeeper.SetParams(suite.ctx, stakingParams)
 }
 
+func init() {
+	config := sdk.GetConfig()
+	cmdcfg.SetBech32Prefixes(config)
+}
```
is the bech32 prefix still falling?

prajjwol replied:
yeah for the clawback test specifically

fedekunze replied:
hmm ok let me look into it

fedekunze reviewed (commented):
minor comments. It'd be great if we could also add tests for `ClawbackEmptyAccounts` and `ClawbackEscrowedTokens` individually. 

- Add tests that check that the claim enabled parameter is true and then disabled after the claim ends
- test that no claim can be performed after ending the claim
- test that the store is empty no ClaimRecords after ending the claim

prajjwol commented:
> minor comments. It'd be great if we could also add tests for `ClawbackEmptyAccounts` and `ClawbackEscrowedTokens` individually.
> 
> * Add tests that check that the claim enabled parameter is true and then disabled after the claim ends
> * test that no claim can be performed after ending the claim
> * test that the store is empty no ClaimRecords after ending the claim

I think I addressed the concerns in `TestClawbackEscrowedTokens` Let me know if I missed something.

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +450,125 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+		},
+		{
+			name:           "non airdrop address active",
+			address:        sdk.AccAddress(tests.GenerateAddress().String()).String(),
```
```suggestion
			address:        sdk.AccAddress(tests.GenerateAddress().Bytes()).String(),
```

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +450,125 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+		err = acc.SetSequence(tc.sequence)
+		suite.Require().NoError(err, "err: %s test: %s", err, tc.name)
+		suite.app.AccountKeeper.SetAccount(suite.ctx, acc)
+		suite.app.ClaimKeeper.SetClaimRecord(suite.ctx, addr, claimRecord)
```
do we want to try using different claim records (i.e with different sets of actions completed)?

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +450,125 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+		suite.Require().NoError(err, "err: %s test: %s", err, tc.name)
+		coins := suite.app.BankKeeper.GetAllBalances(suite.ctx, addr)
+		if tc.expectClawback {
+			suite.Require().True(coins.AmountOfNoDenomValidation(params.GetClaimDenom()).Equal(sdk.ZeroInt()),
```
```suggestion
			suite.Require().Equal(coins.AmountOfNoDenomValidation(params.GetClaimDenom()), sdk.ZeroInt(),
```

fedekunze commented on x/claim/keeper/claim_test.go:
```diff
@@ -454,68 +450,125 @@ func (suite *KeeperTestSuite) TestClaimOfDecayed() {
+			suite.Require().True(coins.AmountOfNoDenomValidation(params.GetClaimDenom()).Equal(sdk.ZeroInt()),
+				"balance incorrect. test: %s", tc.name)
+		} else {
+			suite.Require().True(coins.AmountOfNoDenomValidation(params.GetClaimDenom()).Equal(sdk.NewInt(100)),
```
```suggestion
			suite.Require().Equal(coins.AmountOfNoDenomValidation(params.GetClaimDenom()), sdk.NewInt(100),
```

fedekunze reviewed (approved):
Nice! We can refactor the tests to follow the table driven test format in a follow up PR
